### ISSUE TITLE
Layout/MultilineMethodCallIndentation false positive for method chains inside hash pairs in method arguments

### ISSUE THREAD

**Issue by nagaki** [rubocop/rubocop#14916]
In RuboCop 1.84.2, a false positive was introduced for the Layout/MultilineMethodCallIndentation rule. When a method chain containing
  hash pairs is passed as an argument to another method, nested method chains within those hash values are incorrectly flagged as
  misaligned, despite being correctly indented.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

# Test cases for Layout/MultilineMethodCallIndentation autocorrect indentation.
class TestIndentation
  def test_simpler_case
    @abc = Abc.published
              .where(id: Xyz.select(:id)
                            .joins(:abc_xyz)
                            .where.not(abc_xyz: { id: 123 }))
  end
end
```

## Expected behavior

No offenses found

## Actual behavior

```
% bundle exec rubocop --debug
For /Code/rubocop-mrp: configuration from /Code/rubocop-mrp/.rubocop.yml
Default configuration from /Code/rubocop-mrp/.devbox/virtenv/ruby/gems/rubocop-1.84.2/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Code/rubocop-mrp/test.rb
Loading cache from /.cache/rubocop_cache/5dcde00bfce10134e13c118f75a4a3af211fcfc2/6d7a3b621ca1730e04accd938619e4bdab66cfb1/0af4378c9e3a06ffdb18a87fe8452a06c053af66
C

Offenses:

test.rb:8:29: C: [Correctable] Layout/MultilineMethodCallIndentation: Align .joins with Xyz.select(:id) on line 7.
                            .joins(:abc_xyz)
                            ^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.11477899993769825 seconds
```

## RuboCop version

```
% bundle exec rubocop -V
1.84.2 (using Parser 3.3.10.2, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 3.4.6) [arm64-darwin24]
```

**Comment by nicolasva** [comment]
Problem
A false positive was introduced in RuboCop's Layout/MultilineMethodCallIndentation cop. When a method chain containing hash pairs is passed as an argument to another method, nested method chains within those hash values are incorrectly flagged as misaligned.

Falsely flagged code example:

@abc = Abc.published
.where(id: Xyz.select(:id)
.joins(:abc_xyz) # false positive here
.where.not(abc_xyz: { id: 123 }))
Root cause
The aligned_with_first_line_dot? method in [multiline_method_call_indentation.rb:169]. This guard was necessary for hash literals but also prevented correct dot-alignment detection for parenthesized method arguments.

Solution : 
Two changes in [multiline_method_call_indentation.rb]

aligned_with_first_line_dot? (line 173) — the return false if first_call == node.receiver guard no longer applies when the hash pair is inside a parenthesized method argument.

New method hash_pair_in_method_argument? (line 184) — detects whether the hash pair is inside a parenthesized method call (e.g. .where(id: ...)) to distinguish hash literals from method arguments.

Tests added
5 new tests in [multiline_method_call_indentation_spec.rb]

Nested method chain inside hash pair argument of outer method chain (original bug case)
Same case with safe navigation (&.)
Same case without assignment
Same case with multiple hash pairs
Same case with deep nesting (3 levels)
Result
257 tests passing (0 failures) on both Parser and Prism parsers.

**Comment by sylvain-8422** [comment]
### What was OK before 1.84.1

```
{
  bool: {
    filter: [
      FirstFilterBuilder
        .new(foo, bar)
        .build_filter,

      SecondFilterBuilder
        .new(baz, bae)
        .build_filter
    ]
  }
}
```

### What is considered OK (and auto-corrected to) since 1.84.1 (still the case in 1.85.0)

```
{
  bool: {
    filter: [
      FirstFilterBuilder
      .new(foo, bar)
      .build_filter,

      SecondFilterBuilder
      .new(baz, bae)
      .build_filter
    ]
  }
}
```

### Conclusion

I would argue that **this is a bug introduced in 1.84.1**. 🐞

**Comment by Splines** [comment]
Not sure if the fix is working with version `1.86.0`, since the following code snippet

```rb
  def users
    User.where(id: LectureUserJoin.where(lecture: lectures)
                                  .pluck(:user_id).uniq)
  end

  class TestIndentation
    def test_simpler_case
      @abc = Abc.published
                .where(id: Xyz.select(:id)
                                 .joins(:abc_xyz)
                                                            .where.not(abc_xyz: { id: 123 }))
    end
  end
```

is autocorrected to

```rb
  def users
    User.where(id: LectureUserJoin.where(lecture: lectures)
                   .pluck(:user_id).uniq)  # autocorrected, although not wanted
  end

  class TestIndentation
    def test_simpler_case
      @abc = Abc.published
                .where(id: Xyz.select(:id)
                                 .joins(:abc_xyz)  # not autocorrected, although wanted
                                                            .where.not(abc_xyz: { id: 123 }))  # not autocorrected, although wanted
    end
  end
```

given these layout rules (full file available [here](https://github.com/MaMpf-HD/mampf/blob/ca87645fb982a87abaef9869af14b935ba407499/.config/.rubocop.yml#L1-L142)):

```yaml
Layout/EndAlignment:
  EnforcedStyleAlignWith: variable

Layout/IndentationConsistency:
  EnforcedStyle: indented_internal_methods

Layout/MultilineMethodCallIndentation:
  EnforcedStyle: aligned

Layout/LineLength:
  Max: 100
```

Notice how for the `users` method, the `.pluck` was moved (not wanted). For the `test_simpler_case` method, nothing was even flagged as a problem (wanted behavior: `.select`, `.joins` and `.where` align).

See also [this discussion](https://github.com/rubocop/rubocop/discussions/14991).

---

Unfortunately, this issue persists since the last five minor versions of RuboCop. We'll have to continue to stay on `1.81.7` where it works fine.

**Comment by Splines** [comment]
Related to #15054.
